tests/namespaced.rs

```diff
@@ -885,6 +885,76 @@ mod namespaced {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_svc_hostname() -> anyhow::Result<()> {
+        let mut manager = setup_netns_test!(Shared);
+        let zt = manager.deploy_ztunnel(DEFAULT_NODE).await?;
+        manager
+            .service_builder("server")
+            .addresses(vec![NetworkAddress {
+                network: strng::EMPTY,
+                address: TEST_VIP.parse::<IpAddr>()?,
+            }])
+            .ports(HashMap::from([(80u16, 8080u16)]))
```

Comment:
A service can have two things here: a numeric target port and a named target port (mapping from Kubernetes API). In the named target port case, this would be `(80, 0)`. The intent is the worklaod then has a port map like `80 -> 8080` and we can use that. I think we still need to handle that case here.

Which I guess implies two test cases may be useful here, though I don't mind making one of them a unit test in inbound.rs if preferred (either is fine with me)

Reply:
@howardjohn I don't think we currently have access to the workload port mapping. Prior to this change we'd only care about the target port and could get this directly from the authority header. I think beyond a test case this might require some WDS changes. Could support for named target ports be a TODO or with a quick follow up or is this a blocker?

Reply:
Update - we do have access through the service endpoints. We can lookup endpoint by workload uid

Reply:
@howardjohn Added an additional e2e and the logic to support named target ports